### DISCUSSION TITLE
nextjs-swc(app-router-demo) change language without page reload

### DISCUSSION THREAD

**Discussion by andeksan** [lingui/js-lingui#2032] [General]
Hi, I looked at the nextjs-swc example. And I wonder if it is possible to rework app-router-demo so that after changing the language the page does not reload. Because pages-router-demo works without reloading the page.

**Comment by andrii-bodnar** [comment]
Hi, It is possible to avoid a full-page reload when switching languages, but it's often not worth the added complexity. 

It's recommended to use a full-page reload, especially with RSC. If your language detection is set up correctly, users should only need to switch languages once, so the reload won't be a big issue. 

This approach is simpler and more reliable in terms of internal state handling, revalidation of loaded data, etc.

**Reply by andeksan** [reply]
> Hi, It is possible to avoid a full-page reload when switching languages, but it's often not worth the added complexity.
> 
> It's recommended to use a full-page reload, especially with RSC. If your language detection is set up correctly, users should only need to switch languages once, so the reload won't be a big issue.
> 
> This approach is simpler and more reliable in terms of internal state handling, revalidation of loaded data, etc.

if it possible, how to implement it in app-router-demo project? any hints.
Any page reload results in additional requests and loss of user input. I want to avoid that. So it's a problem for me.

**Comment by timofei-iatsenko** [comment]
Agree with the Andrii, but answering directly to your queustion, simply use nextjs router methods to switch the page to other language and you will get a desired behaviour.

**Reply by andeksan** [reply]
> Agree with the Andrii, but answering directly to your queustion, simply use nextjs router methods to switch the page to other language and you will get a desired behaviour.

you mean page router?! i want to avoid page router because, this will lead to a huge reworking of the infrastructure.
so i want to focus on app router.

**Reply by timofei-iatsenko** [reply]
https://nextjs.org/docs/app/api-reference/functions/use-router
